File: templates/template.hs
{-
 - ExX_X.hs
 - Kevin C. Baird
 - SICP in Haskell
 -
 -}

aPlusAbsB :: Int -> Int -> Int
aPlusAbsB a b = a `op` b
  where
    op
      | b > 0     = (+)
      | otherwise = (-)

File: templates/template.rb
#!/usr/bin/env ruby
# XXX.rb

=begin rdoc
Kevin C. Baird
SICP in Ruby
XXX
=end

def a_plus_abs_b(a, b)
  op = b > 0 ? :+ : :-
  a.send(op, b)
end
